test/integration_tests.jl

```diff
@@ -152,3 +140,29 @@
 
     _cleanup_old_branches(url)
 end
+
+@testset "$(service)" for service in [GITHUB, GITLAB]
+    personal_access_token = ENV["INTEGRATION_PAT_$(service)"]
+    test_repo = ENV["INTEGRATION_TEST_REPO_$(service)"]
+
+    env = Dict(
+        "$(service)_REPOSITORY" => test_repo,
+        "$(service)_TOKEN" => personal_access_token,
+        "CI_PROJECT_PATH" => ENV["INTEGRATION_TEST_REPO_GITLAB"]
+    )
+
+    if service == GITHUB
+        env["GITHUB_REPOSITORY"] = "true"
+    elseif service == GITLAB
+        env["GITLAB_CI"] = "true"
+    end
+
+    ci_cfg = CompatHelper.auto_detect_ci_service(env=env)
+    hostname = CompatHelper.clone_hostname(ci_cfg)
+
+    api, repo = CompatHelper.get_api_and_repo(ci_cfg, hostname; env=env)
```

Comment:
this line need the `api_hostname` because it's getting the api, and the "get_url_with_auth" needs the `clone_hostname` since it's just generating a url for cloning. So you'll need to grab both `CompatHelper.clone_hostname` and `CompatHelper.api_hostname`